src/RevCPU.cc

```diff
@@ -267,25 +267,47 @@ RevCPU::RevCPU( SST::ComponentId_t id, const SST::Params& params )
   // Initial thread setup
   uint32_t MainThreadID = id+1; // Prevents having MainThreadID == 0 which is reserved for INVALID
 
-  // set the pc
+
   uint64_t StartAddr = 0x00ull;
-  std::string StartSymbol = "main";
-  if( StartAddr == 0x00ull ){
-    // if( !Opts->GetStartSymbol( id, StartSymbol ) ){
-    //   output.fatal(CALL_INFO, -1,
-    //                 "Error: failed to init the start symbol address for main thread=\n");
-    // }
-    StartAddr = Loader->GetSymbolAddr(StartSymbol);
-  }
-  if( StartAddr == 0x00ull ){
-    // load "main" symbol
-    StartAddr = Loader->GetSymbolAddr("main");
-    if( StartAddr == 0x00ull ){
-      output.fatal(CALL_INFO, -1,
-                   "Error: failed to auto discover address for <main> for main thread\n");
+  std::string StartSymbol;
+
+  bool IsStartSymbolProvided = Opts->GetStartSymbol( id, StartSymbol );
+  bool IsStartAddrProvided = Opts->GetStartAddr( id, StartAddr ) && StartAddr != 0x00ull;
+  uint64_t ResolvedStartSymbolAddr = (IsStartSymbolProvided) ? Loader->GetSymbolAddr(StartSymbol) : 0x00ull;
+
+  // If no start address has been provided ...
+  if (!IsStartAddrProvided) {
+    // ... check if symbol was provided ...
+    if (!IsStartSymbolProvided) {
+        // ... no, try to default to 'main' ...
+        StartAddr = Loader->GetSymbolAddr("main");
+        if( StartAddr == 0x00ull ){
+          // ... no hope left!
+          output.fatal(CALL_INFO, -1,
+                       "Error: failed to auto discover address for <main> for main thread\n");
+        }
+    } else {
+      // ... if symbol was provided, check whether it is valid or not ...
+      if (!ResolvedStartSymbolAddr) {
+        // ... not valid, error out
+        output.fatal(CALL_INFO, -1,
+                  "Error: failed to resolve address for symbol <%s> for core=%lu\n", StartSymbol.c_str(), id);
+      }
+      // ... valid use the resolved symbol
+      StartAddr = ResolvedStartSymbolAddr;
     }
+  } else { // A start address was provided ...
+    // ... check if a symbol was provided and is compatible with the start address ...
+    if ((IsStartSymbolProvided) && (ResolvedStartSymbolAddr != StartAddr)) {
+      // ... they are different, don't know the user intent so error out now
+      output.fatal(CALL_INFO, -1,
+                  "Error: start address and start symbol differ for core=%lu startAddr=%lu StartSymbol=%s ResolvedStartSymbolAddr=%lu\n",
+                    id, StartAddr, StartSymbol.c_str(), ResolvedStartSymbolAddr);
+    } // ... else no symbol provided, continue on with StartAddr as the target
```

Comment:
Can you modify your outputs to use the formatting macros (`PRIu64`, `PRIu32`, `PRIx64` etc.)?

```cpp
output.fatal(CALL_INFO, -1,
                  "Error: start address and start symbol differ for core=%" PRIu32 " startAddr=%" PRIx64 " StartSymbol=%s ResolvedStartSymbolAddr=%" PRIu64 "\n",
                    id, StartAddr, StartSymbol.c_str(), ResolvedStartSymbolAddr);
```

Reply:
fixed